src/openads/domain/service/DomainEventBus.js

```diff
@@ -12,11 +12,28 @@ export default class DomainEventBus {
     }
     if (!this._observers.has(eventName)) {
       this._observers.set(eventName, [observer])
+    } else {
+      this._observers.get(eventName).push(observer)
     }
-    this._observers.get(eventName).push(observer)
   }
 
-  static raise ({domainEvent}) {
-    this._observers.get(domainEvent.eventName).forEach(observer => observer({payload: domainEvent.payload}))
+  raise ({domainEvent}) {
+    this._observers
+      .get(domainEvent.eventName)
+      .forEach(observer => observer({
```

Comment:
Just a question:
The observers will execute in order sequentially and if one fails we will face that all the observers chain will fail. Is that the expected behavior?

Reply:
You are right. We need a try catch block wrapping the observer to not stop execution. 
Also in the catch block will be great have some kind of log...

Reply:
Done

![](https://media.giphy.com/media/3oFzmqN1xHwaEXGl7q/giphy.gif)